src/@ionic-native/plugins/purchases/index.ts

```diff
@@ -527,6 +527,81 @@ export class Purchases extends IonicNativePlugin {
   ): Promise<{ [productId: string]: IntroEligibility; }> {
     return;
   }
+
+  /**
+   * Sets a function to be called on purchases initiated on the Apple App Store. This is only used in iOS.
+   * @param {ShouldPurchasePromoProductListener} shouldPurchasePromoProductListener Called when a user initiates a
+   * promotional in-app purchase from the App Store. If your app is able to handle a purchase at the current time, run
+   * the deferredPurchase function. If the app is not in a state to make a purchase: cache the deferredPurchase, then
+   * call the deferredPurchase when the app is ready to make the promotional purchase.
+   * If the purchase should never be made, you don't need to ever call the deferredPurchase and the app will not
+   * proceed with promotional purchases.
+   */
+  @Cordova()
```

Comment:
I think all these new functions should be `Cordova({ sync: true })` since they are not returning Observable/Promise. https://github.com/RevenueCat/ionic-native/blob/master/DEVELOPER.md#cordova

Reply:
great catch 👌